src/autoscaler/eventgenerator/server/server_test.go

```diff
@@ -0,0 +1,82 @@
+package server_test
+
+import (
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const TestPathAggregatedMetricHistories = "/v1/apps/an-app-id/aggregated_metric_histories/a-metric-type"
+
+var _ = Describe("Server", func() {
+	var (
+		rsp *http.Response
+		err error
+	)
+
+	Context("when retrieving aggregared metrics history", func() {
+		BeforeEach(func() {
+			serverUrl.Path = TestPathAggregatedMetricHistories
+		})
+
+		JustBeforeEach(func() {
+			rsp, err = http.Get(serverUrl.String())
+		})
+
+		It("should return 200", func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rsp.StatusCode).To(Equal(http.StatusOK))
+			rsp.Body.Close()
+		})
+	})
+
+	Context("when requesting the wrong path", func() {
+		BeforeEach(func() {
+			serverUrl.Path = "/not-exist-path"
+		})
+
+		JustBeforeEach(func() {
+			rsp, err = http.Get(serverUrl.String())
+		})
+
+		It("should return 404", func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rsp.StatusCode).To(Equal(http.StatusNotFound))
+			rsp.Body.Close()
+		})
+	})
+
+	Context("when using wrong method to retrieve aggregared metrics history", func() {
+		BeforeEach(func() {
+			serverUrl.Path = TestPathAggregatedMetricHistories
+		})
+
+		JustBeforeEach(func() {
+			rsp, err = http.Post(serverUrl.String(), "garbage", nil)
+		})
+
+		It("should return 404", func() {
+			Expect(err).ToNot(HaveOccurred())
+			Expect(rsp.StatusCode).To(Equal(http.StatusNotFound))
+			rsp.Body.Close()
+		})
+	})
+
+	Context("when using wrong method to retrieve aggreated metric histories", func() {
```

Comment:
what is the difference between this one and the one above?

Reply:
it is a duplicate, will remove